manifests/init.pp

```diff
@@ -146,18 +147,20 @@
 
   include foreman_proxy_content::pub_dir
 
-  if $qpid_router {
-    class { 'foreman_proxy_content::dispatch_router':
-      agent_addr    => $qpid_router_agent_addr,
-      agent_port    => $qpid_router_agent_port,
-      ssl_ciphers   => $qpid_router_ssl_ciphers,
-      ssl_protocols => $qpid_router_ssl_protocols,
-      logging_level => $qpid_router_logging_level,
-      logging       => $qpid_router_logging,
-      logging_path  => $qpid_router_logging_path,
-    }
-    contain foreman_proxy_content::dispatch_router
+  class { 'foreman_proxy_content::dispatch_router':
+    agent_addr     => $qpid_router_agent_addr,
+    agent_port     => $qpid_router_agent_port,
+    ssl_ciphers    => $qpid_router_ssl_ciphers,
+    ssl_protocols  => $qpid_router_ssl_protocols,
+    logging_level  => $qpid_router_logging_level,
+    logging        => $qpid_router_logging,
+    logging_path   => $qpid_router_logging_path,
+    service_ensure => $enable_katello_agent,
+    service_enable => $enable_katello_agent,
+  }
+  contain foreman_proxy_content::dispatch_router
```

Comment:
The only concern I have here is that this now always installs the router packages, even if the service is then disabled. Previously it didn't even install the packages.

Reply:
Do we actually gain anything by having

```
class { 'foreman_proxy_content::dispatch_router':
  ...
}
contain foreman_proxy_content::dispatch_router
```

Declared outside of the

```
if $enable_katello_agent {
  ...
}
```

block?

It's not clear to me why you are moving it here.

Reply:
It exists outside of the `enable_katello_agent` block because we don't know the previous state, and therefore we want to ensure the service isn't running if it previously was running. If we were to always put it within the `enable_katello_agent` block you could enable it and then disable it while leaving the services running.

The question I posed in a different forum, which @ekohl asks here (https://github.com/theforeman/puppet-foreman_proxy_content/pull/329#discussion_r605534618) is whether it is sufficient to just disable the services when disabling katello-agent support (or when freshly setting things up to have no presence of) or if we need to propagate this effect to all entities: services, packages, config files. The same question applies here: https://github.com/theforeman/puppet-katello/pull/387/files#diff-4d995f8e71429e47718f73b6c9d8cad415c63a67017803143f5fa1a929d0f730R19